### PR TITLE
refine user verification and authz gesture definitions, add Test of User Presence

### LINKED ISSUE #357: User Verification definition needs to be refined
As per WD-04:  https://www.w3.org/TR/webauthn/#terminology, the definition of User Verification does not address whether the verification takes place within the authenticator boundary.  Recommend adding the following sentence immeidately after 'Note that invocation of said operations implies use of key material managed by the authenticator':  
"Therefore the user verification operation must take place within the restricted operating environment of the authenticator itself."

### PR DESCRIPTION
**Pull request by equalsJeffH**
fixes #357, improves #358 -- see below for more details

### CODE REVIEW

**Comment by equalsJeffH** [comment]
this PR refines/clarifies the definitions of:

authorization-gesture
- Ceremony
- User Consent
- user verification

adds definitions for:
- Biometric Recognition
- Test of User Presence

as well as adding some autolinks where appropriate.

File: index.bs [hunk 1]
```diff
@@ -230,17 +230,25 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
-    that carry data. What is out-of-band to a protocol is in-band to a ceremony. In this specification, <a>Registration</a>,
-    <a>Authentication</a>, and <a>user verification</a> are ceremonies.
+: <dfn>Biometric Recognition</dfn
+:: The automated recognition of individuals based on their biological and behavioral characteristics [[ISOBiometricVocabulary]]. 
 
+: <dfn>Ceremony</dfn>
+:: The concept of a [=ceremony=] [[Ceremony]] is an extension of the concept of a network protocol, with human nodes alongside
+    computer nodes and with communication links that include user interface(s), human-to-human communication, and transfers of
+    physical objects that carry data. What is out-of-band to a protocol is in-band to a ceremony. In this specification,
+    <a>Registration</a> and <a>Authentication</a> are ceremonies, and an <a>authorization gesture</a> is often a component of
```
Comment (jcjones) [inline index.bs]: Nit: The gesture is always a component of those ceremonies, right? The definition of user consent says the gesture  is _the_ ceremony component that defines consent.

File: index.bs [hunk 2]
```diff
@@ -284,19 +292,29 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
-    <a>authenticatorMakeCredential</a> and <a>authenticatorGetAssertion</a> operations, for example through a touch plus pin
-    code, a password, a gesture (e.g., presenting a fingerprint), or other modality. Note that invocation of said operations
-    implies use of key material managed by the authenticator.
+:: The technical process by which an <a>authenticator</a> <em>locally authorizes</em> the invocation of the
+    <a>authenticatorMakeCredential</a> and <a>authenticatorGetAssertion</a> operations. [=User verification=] may be instigated
+    through various [=authorization gesture=] modalities; for example, through a touch plus pin code, password entry, or
+    [=biometric recognition=] (e.g., presenting a fingerprint) [[ISOBiometricVocabulary]]. Note that invocation of the
+    <a>authenticatorMakeCredential</a> and <a>authenticatorGetAssertion</a> operations implies use of key material managed by
+    the authenticator. Note that for security, [=user verification=] and use of [=credential private keys=] must occur within a
+    single logical security boundary defining the [=authenticator=].
```
Comment (vijaybh) [inline index.bs]: Should we also add a sentence saying that a "fresh enough" user verification can also implicitly provide a TUP?

Comment (jcjones) [inline index.bs]: Hmm... would we need to provide normative parameters for how to measure the freshness of the user check?

Comment (equalsJeffH) [inline index.bs]: yeah, I did not as yet go into the so-called "touchless/silent" authentication aka getAssertion topic -- add that in a separate PR?

File: index.bs (hunk 1 above)
Comment (equalsJeffH) [inline index.bs]: @jcjones thx for your review.

actually, it says..
> ...an authorization gesture is _**often a component of**_
    those [=ceremonies=].

Does that address your concern?

File: index.bs [hunk 3]
```diff
@@ -284,19 +292,29 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
 
 : <dfn>User Verification</dfn>
-:: The process by which an <a>authenticator</a> <em>locally authorizes</em> the invocation of the
-    <a>authenticatorMakeCredential</a> and <a>authenticatorGetAssertion</a> operations, for example through a touch plus pin
-    code, a password, a gesture (e.g., presenting a fingerprint), or other modality. Note that invocation of said operations
-    implies use of key material managed by the authenticator.
+:: The technical process by which an <a>authenticator</a> <em>locally authorizes</em> the invocation of the
+    <a>authenticatorMakeCredential</a> and <a>authenticatorGetAssertion</a> operations. [=User verification=] may be instigated
+    through various [=authorization gesture=] modalities; for example, through a touch plus pin code, password entry, or
+    [=biometric recognition=] (e.g., presenting a fingerprint) [[ISOBiometricVocabulary]]. Note that invocation of the
```
Comment (vijaybh) [inline index.bs]: Based on discussion on today's call, maybe we should add a sentence saying that the intent is to be able to tell different human users apart by some means.

Comment (equalsJeffH) [inline index.bs]: done in https://github.com/w3c/webauthn/pull/365/commits/d8b6c899a0db6d92530f044071a5e257f3ad3adf

**Comment by equalsJeffH** [comment]
I believe @vijaybh and @jcjones comments are now addressed -- re-review?  thx.

File: index.bs (hunk 1 above)
Comment (jcjones) [inline index.bs]: No, but it's a nit.

Basically, if/when we permit the user consent to be obtained at a time before conducting a ceremony (a "recent consent" or "cached consent" perhaps), we'd need to update the definition of either "ceremony" or "user consent" to account for things happening not at the same time.

Right now, the statement 
>    ...an authorization gesture is often a component of those [=ceremonies=].

is flexible whereas the other definitions are not, so it seems to me it _could_ be rewritten as:
>     ...an authorization gesture is a component of those [=ceremonies=].

But I approve this PR as is, there's no need to an update on this account.

Comment (equalsJeffH) [inline index.bs]: Ok, have also update the dfn for user consent:
> An [=authorization gesture=] is a [=ceremony=] component often employed to indicate [=user consent=].

how's that?

**Comment by jcjones** [comment]
Thanks, 7b85ee10823e6b3453194a438ba908258f03ae2f looks perfect.

**Comment by vijaybh** [comment]
LGTM, merging.

**Pull request merged**